runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/control/FnApiControlClientPoolService.java

```diff
@@ -34,13 +34,14 @@
     implements FnService {
   private static final Logger LOGGER = LoggerFactory.getLogger(FnApiControlClientPoolService.class);
 
-  private final BlockingQueue<FnApiControlClient> clientPool;
+  private final ThrowingConsumer<? super FnApiControlClient> clientPool;
   private final Collection<FnApiControlClient> vendedClients = new CopyOnWriteArrayList<>();
   private final HeaderAccessor headerAccessor;
   private AtomicBoolean closed = new AtomicBoolean();
 
   private FnApiControlClientPoolService(
-      BlockingQueue<FnApiControlClient> clientPool, HeaderAccessor headerAccessor) {
+      ThrowingConsumer<? super FnApiControlClient> clientPool,
```

Comment:
I tried to rename this, but it made for a surprisingly annoying time when trying to rebase this into the correct commit. I don't think it's worth making that change here because this file is going away anyway.